dataframely/functional.py

```diff
@@ -23,41 +23,86 @@
 
 # --------------------------------- RELATIONSHIP 1:1 --------------------------------- #
 
+LEN_COLUMN = "__len__"
 
-def filter_relationship_one_to_one(
+
+def require_relationship_one_to_one(
     lhs: LazyFrame[S] | pl.LazyFrame,
     rhs: LazyFrame[T] | pl.LazyFrame,
     /,
     on: str | list[str],
+    *,
+    filter_unique: bool = True,
 ) -> pl.LazyFrame:
     """Express a 1:1 mapping between data frames for a collection filter.
 
     Args:
         lhs: The first data frame in the 1:1 mapping.
         rhs: The second data frame in the 1:1 mapping.
-        on: The columns to join the data frames on. If not provided, the join columns
-            are inferred from the mutual primary keys of the provided data frames.
+        on: The columns to join the data frames on.
+        filter_unique: If set to `True`, drops rows that are not uniquely identified by the
+            join columns specified with `on`. If set to `False`, skips uniqueness checks
+            and avoids performance penalties. Use with caution, as this may lead to unexpected
+            results if rows in one or both of the data frames are not unique in the join columns.
+
+    Returns:
+        A data frame representing the inner join of the two inputs on the specified
+        columns, filtered to ensure a 1:1 relationship.
     """
+    if filter_unique:
+        return (
+            lhs.group_by(on)
+            .len(LEN_COLUMN)
+            .filter(pl.col(LEN_COLUMN) == 1)
+            .drop(LEN_COLUMN)
```

Comment:
Why are we not using `unique` here? 👀 
```suggestion
            lhs.unique(on)
```

Reply:
Ah yes, absolutely, we need the `keep="none"`!